File: hahahaeun/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []

    for i in range(len(s)):
        if s[i] not in s[:i]:
            answer.append(-1)
        else:
            for j in range(i-1, -1, -1):
                if s[j] == s[i]:
                    answer.append(i-j)
                    break        
    return answer

File: yetpi0413/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    map<char, int> dic;
    for(int i=0; i<s.length(); i++){
        if(dic.find(s[i]) != dic.end()){
            answer.push_back(i-dic[s[i]]);
        }
        else{
            answer.push_back(-1);
        }
        dic[s[i]] = i;
    }
    return answer;
}